### ISSUE TITLE
disable access to control routes via 192.168.127.1

### ISSUE THREAD

**Issue by protosam** [containers/gvisor-tap-vsock#78]
Using the podman machine, containers can access all the controls via `curl -s 192.168.127.1/...`. This behavior should probably be disabled or gated it some way.

It has security implications if end users decide to use podman to host containerized applications on Mac for example, which is something that absolutely happens with Docker for Desktop on Mac.

**Comment by guillaumerose** [comment]
Yes good idea. We could do that with a certificate that can be mounted in the VM.

**Comment by protosam** [comment]
What do you think about adding functionality to create unix sockets on VMs that route to services? 

Thinking I would like to increase the level of effort to access this to be file system access (or container escape).

**Comment by guillaumerose** [comment]
> What do you think about adding functionality to create unix sockets on VMs that route to services?

It would imply something new in the VM to handle that no? A process or a good systemd magic configuration?

**Comment by protosam** [comment]
Could probably take advantage of an ssh client similar to what is being done for unix2unix sockets in the forwarder service.

**Comment by MrEcco** [comment]
I have enabled a more flexible way to configure gvproxy in my PR: https://github.com/containers/gvisor-tap-vsock/pull/484. It allows changing the stack configuration, for example disabling host forwarding into userspace network.

I would appreciate it if you vote with 👍🏻 in the PR :) I hope it may fasten the merge.
